### PR TITLE
[Refactor] DCR Monitor Refactoring (1/N)

### PR DESCRIPTION
**Pull request by dayeol**
This is a first PR for a major refactoring of DCR monitor. See #15

> [!WARNING] 
> This PR breaks attestation functionality, because `TEEProvider` does not implement attestation feature yet.

# Changes
* Change Monitor to be a "Reconciler", which runs a reconciliation loop every 10 seconds. This required me to change k8s `CronJob` to `Deployment`, and have reconciliation loop in the app (cronjob only supports > 1 minute)
* Create `TEEProvider` type to support different TEE backends. Only partially implemented right now. 
* Clean up unused functions (`ListAllInstances`, `DeleteInstance`)
* Added a unit test for `reconcileJob` function
* Added the `bazel test //...` to the CI

# Testing

Manually deployed & tested. Also added a unit test for `updateJobStatus`
didn't test `Reconcile`, it needs dependency injection for the DB.

### CODE REVIEW

File: app/dcr_monitor/main.go [hunk 1]
```diff
@@ -3,28 +3,55 @@ package main
...
 	"github.com/cloudwego/hertz/pkg/common/hlog"
+	"github.com/manatee-project/manatee/app/dcr_api/biz/dal/db"
 	"github.com/manatee-project/manatee/app/dcr_monitor/client"
 	"github.com/manatee-project/manatee/app/dcr_monitor/monitor"
 	"github.com/manatee-project/manatee/pkg/config"
 )
 
 func main() {
+	// Set up HTTP server for liveness probe
+	http.HandleFunc("/healthz", healthHandler)
```
Comment (xxxxavier) [inline app/dcr_monitor/main.go:18]: Why do we need this part?

File: app/dcr_monitor/reconciler.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+				return fmt.Errorf("failed to get instance status: %w", err)
+			}
+			if instanceStatus == "TERMINATED" {
+				j.JobStatus = int(job.JobStatus_VMFinished)
+			}
+		}
+	}
+
+	// clean up instance if necessary
+	if j.JobStatus == int(job.JobStatus_VMFinished) || j.JobStatus == int(job.JobStatus_VMFailed) {
```
Comment (xxxxavier) [inline app/dcr_monitor/reconciler.go:90]: Should we do clean up after we update job status

File: app/dcr_monitor/main.go (hunk 1 above)
Comment (dayeol) [reply app/dcr_monitor/main.go:18]: actually we do not need it. I didn't remove it after debugging.

File: app/dcr_monitor/reconciler.go (hunk 2 above)
Comment (dayeol) [reply app/dcr_monitor/reconciler.go]: good point, fixed!

**Review by xxxxavier — APPROVED** [review]

**Review by privacywill — APPROVED** [review]

**Pull request merged**